### PR TITLE
disallow empty static node names

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/openshift/origin/issues/1751.

@liggitt

### CODE REVIEW

File: pkg/cmd/server/start/master_args.go [hunk 1]
```diff
@@ -338,10 +338,13 @@ func (args MasterArgs) BuildSerializeableKubeMasterConfig() (*configapi.Kubernet
 		masterIP = ip.String()
 	}
 
+	staticNodeList := util.NewStringSet(args.NodeList...)
```
Comment (liggitt) [inline pkg/cmd/server/start/master_args.go]: `--nodes=` resulting in `[""]` seems like a bug in `util.StringList`

File: pkg/cmd/server/api/validation/master.go [hunk 2]
```diff
@@ -144,6 +145,12 @@ func ValidateKubernetesMasterConfig(config *api.KubernetesMasterConfig) fielderr
 		allErrs = append(allErrs, ValidateFile(config.SchedulerConfigFile, "schedulerConfigFile")...)
 	}
 
+	for i, nodeName := range config.StaticNodeNames {
+		if len(nodeName) == 0 {
+			allErrs = append(allErrs, fielderrors.NewFieldInvalid(fmt.Sprintf("staticNodeName[%d]", i), config.StaticNodeNames[i], "may not be empty"))
```
Comment (liggitt) [inline pkg/cmd/server/api/validation/master.go]: use `nodeName`, not `config.StaticNodeNames[i]`

**Comment by liggitt** [comment]
nits, and I'd prefer an upstream fix in StringList, but LGETM (looks good enough to merge)

**Comment by deads2k** [comment]
[merge]

**Pull request merged**